### PR TITLE
[SIG-1881] useFetchUser hook

### PR DESCRIPTION
**Pull request by janjaap**
This PR adds a `useFetchUser` hook that retrieves user detail data on component mount.

### CODE REVIEW

File: src/signals/settings/users/containers/Detail/__tests__/Detail.test.js [hunk 1]
```diff
@@ -1,19 +1,39 @@
...
+    await act(async () => {
+      rerender(withAppContext(<UserDetail location={{ referrer }} />));
+    });
 
     expect(container.querySelector('a').getAttribute('href')).toEqual(referrer);
   });
+
+  it('should show an alert on error', async () => {
+    let getByText;
+    const error = new Error('Not allowed');
```
Comment (spaces-ga) [inline src/signals/settings/users/containers/Detail/__tests__/Detail.test.js]: Could we also test this message `toBeInTheDocument()`?

File: src/signals/settings/users/containers/Detail/hooks/useFetchUser.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        setError(e);
+      } finally {
+        setLoading(false);
+      }
+    }
+
+    fetchData();
+
+    return (() => {
+      controller.abort();
```
Comment (spaces-ga) [inline src/signals/settings/users/containers/Detail/hooks/useFetchUser.js]: I thought you added an `unmount` test as well... 🤔

File: src/signals/settings/users/containers/Detail/index.js [hunk 3]
```diff
@@ -1,28 +1,57 @@
...
+              <br />
+              Voornaam: {data.first_name}
+              <br />
+              Achternaam: {data.last_name}
+              <br />
+              Gebruikersnaam: {data.username}
+            </Fragment>
+          )}
+        </Column>
+        <Column span={7}></Column>
```
Comment (spaces-ga) [inline src/signals/settings/users/containers/Detail/index.js]: Curious what this `Column` is for, but also why this file is in this particular PR. Looks like it belongs in a different PR.

Comment (janjaap) [reply src/signals/settings/users/containers/Detail/index.js]: I've created this file to be able to test the hook. It will be populated in another PR (#464)

File: src/signals/settings/users/containers/Detail/__tests__/Detail.test.js (hunk 1 above)
Comment (janjaap) [reply src/signals/settings/users/containers/Detail/__tests__/Detail.test.js]: The message itself won't be in the document

File: src/signals/settings/users/containers/Detail/hooks/useFetchUser.js (hunk 2 above)
Comment (janjaap) [reply src/signals/settings/users/containers/Detail/hooks/useFetchUser.js]: That was another hook. But well spotted; I'll add that test as well.
